gradio/block_function.py

```diff
@@ -0,0 +1,175 @@
+from __future__ import annotations
```

Comment:
I did this to prevent a circular import

Reply:
Interesting. shouldn't be strictly necessary, but not a bad idea to break up blocks.py